feat(components): document adding lint rules to desktop modeler

philippfromme opened the pull request:
Related to https://github.com/camunda/camunda-modeler/issues/3269

## What is the purpose of the change

Camunda Modeler 5.4 supports adding lint rules for the new built-in linting feature (introduced in Camunda Modeler 5.0). Lint rules can be added through a Camunda Modeler plugin which is [already documented](https://docs.camunda.io/docs/next/components/modeler/desktop-modeler/plugins/) but since it's a special kind of plugin we want to add a dedicated section for it. That section is also [linked in the general plugins section](https://docs.camunda.io/docs/next/components/modeler/desktop-modeler/plugins/#overview-of-your-possibilities-as-a-plugin-developer).

## Are there related marketing activities

No further activities.

## When should this change go live?

Immediately, as this feature has already been released.

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [ ] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

barmac commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+
+Take the following steps:
+
+1. Clone the repository:
```
If we want to encourage people to share their work, I'd rather suggest to fork the repo instead.

nikku commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+---
+id: lint-rules
+title: Lint rules
+description: "Lint rule plugins allow you to add your own lint rules to the Desktop Modeler."
```
```suggestion
description: "Learn how to contribute additional validation (lint) rules to the Desktop Modeler through lint rule plug-ins."
```

nikku commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+description: "Lint rule plugins allow you to add your own lint rules to the Desktop Modeler."
+---
+
+[bpmnlint](https://github.com/bpmn-io/bpmnlint) plugins allow you to add custom lint rules for BPMN diagrams or disable existing rules.
```
```suggestion
The Desktop Modeler uses [bpmnlint](https://github.com/bpmn-io/bpmnlint) to validate BPMN diagrams. Through defined modeler extension points you add plug-in additional lint rules or configure/disable existing rules.
```

nikku commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+
+## Getting started
+
+To get started, you can use [this example project](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin).
```
```suggestion
To get started, you can start from the [custom-linter-rules-plugin template)[https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin).
```

nikku reviewed (commented):
Some cosmetic suggestions from my end; looks nice! :)

philippfromme commented:
This pull request has been reviewed internally in the Desktop Modeler and bpmn.io team. It's ready to be reviewed and merged. @akeller @christinaausley @pepopowitz 👀

christian-konrad commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+---
+id: lint-rules
+title: Lint rules
```
What about "Custom lint rules"?

christian-konrad commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+description: "Learn how to contribute additional validation (lint) rules to the Desktop Modeler through lint rule plug-ins."
+---
+
+The Desktop Modeler uses [bpmnlint](https://github.com/bpmn-io/bpmnlint) to validate BPMN diagrams. Through defined modeler extension points you add plug-in additional lint rules or configure/disable existing rules.
```
Typo "you add plug-in additional..."

christian-konrad replied:
Also, better to highlight the value of this feature first (allows to add your own validation rules to ensure processes follow blabla...) before diving into the technical details

christinaausley replied:
Reworded this in my latest commit -- is this what you mean?

christian-konrad replied:
@philippfromme @christinaausley more like explaining what those rules are for—why should a user care about it? Maybe also adding a screenshot of how a custom rule will appear in the problems panel, making it easier to correlate for the reader

philippfromme replied:
I was also thinking about adding a screenshot to sell it better. Will make adjustments. I'll also try and discuss this again with @nikku.

christian-konrad commented on docs/components/modeler/desktop-modeler/lint-rules/lint-rules.md:
```diff
@@ -0,0 +1,82 @@
+
+3. Add a custom rule
+
+To add a custom rule, add it to the [`bpmnlint-plugin-custom/rules`](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin/tree/master/bpmnlint-plugin-custom/rules) folder. The example project contains a [`no-manual-task.js`](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin/blob/master/bpmnlint-plugin-custom/rules/no-manual-task.js) file implements a custom ru ...
```
Typo: "... file THAT implements a custom rule"

christinaausley replied:
Noticed this as well -- adjusted in my latest commit.

philippfromme commented:
> Let me know if you would like a technical review here 👍

Will let you know after addressing @christian-konrad's comments.

christinaausley reviewed (dismissed):
Adjusted some typos, spacing, and the numerical list in my latest commit. Let me know if you have any questions about these changes.

philippfromme commented:
@christinaausley Just to clarify, this PR is not ready yet. I want to address @christian-konrad's comments first. 😃

christinaausley commented:
@philippfromme Apologies for the premature adjustments -- I saw you had mentioned it's ready to be reviewed and merged and I pushed my review at just about the same time you made the comment about @christian-konrad's comments first. Looks like we may need to merge main in here, but let me know when you'd like me to take a look 👍

philippfromme commented:
@christinaausley no worries. I've made a few adjustments:

* rephrased the introduction to introduce as little new terminology as possible
* renamed the page to _Custom lint rules_
* added a screenshot (will be automated as part of https://github.com/camunda/camunda-docs-modeler-screenshots/ later)

It's now ready for your final review. 🚀

nikku reviewed (approved):
Looks great + reads great to me.

akeller commented:
> Immediately, as this feature has already been released.

Quick question here - was this released prior to the docs being written on purpose or by accident?

nikku commented:
> was this released prior to the docs being written on purpose or by accident?

On purpose. We've already documented the actual thing [on GitHub](https://github.com/camunda/camunda-modeler-custom-linter-rules-plugin) and simply follow up inside the camunda.io docs now.

philippfromme commented:
@christinaausley Are we allowed to merge?

akeller commented:
@philippfromme yes please merge when ready! We can even do a release today so the changes are live.

philippfromme commented:
Sorry @christinaausley and @akeller, can I get another approval? I rebased the branch on master and squashed the commits.

philippfromme commented:
🚀🚀🚀
